altair/utils/html.py

```diff
@@ -13,17 +15,29 @@
         font-weight: normal;
         font-size: 13px;
     }}
+    .error {{
+        color: red;
+    }}
   </style>
   <script src="{base_url}/vega@{vega_version}"></script>
   <script src="{base_url}/vega-lite@{vegalite_version}"></script>
   <script src="{base_url}/vega-embed@{vegaembed_version}"></script>
 </head>
 <body>
-  <div id="vis"></div>
+  <div id="{output_div}"></div>
   <script type="text/javascript">
     var spec = {spec};
     var opt = {embed_opt};
-    vegaEmbed("#vis", spec, opt);
+    vegaEmbed("#{output_div}", spec, opt)
+      .catch(function(error){{
+        var outputDiv = document.getElementById('{output_div}');
+        outputDiv.innerHTML = ('<div class="error">'
+                               + '<p>Javascript Error: ' + error.message + '</p>'
+                               + "<p>This usually means there's a typo in your chart specification. "
+                               + "See the javascript console for the full traceback</p>"
+                               + '</div>');
+        console.error(error);
```

Comment:
I think it would be better to re-throw the error instead. 

```js
throw error;
```

This way you get the full stack trace.